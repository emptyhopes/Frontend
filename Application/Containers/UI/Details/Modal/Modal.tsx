import styled from "styled-components";

import { IoCloseCircleOutline } from "react-icons/io5";

import { PropsInterface } from "@/Application/Containers/UI/Details/Modal/Types/PropsInterface";

const StyledModal = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  width: 100vw;
  height: 100vh;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.8);
  opacity: 0;
  pointer-events: none;
  transition: 0.5s;

  &.active {
    opacity: 1;
    pointer-events: all;
  }
`;

const StyledContent = styled.div`
  position: relative;
  width: 800px;
  height: 600px;
  border: 1px solid #000000;
  border-radius: 20px;
  background-color: #ffffff;
`;

const StyledButton = styled.button`
  position: absolute;
  top: 12px;
  right: 10px;
  width: 30px;
  height: 30px;
  border: none;
  background: transparent;

  & svg {
    width: 20px;
    height: 20px;
  }
`;

const Modal: React.FunctionComponent<PropsInterface> = (props: PropsInterface): React.ReactElement => {
  return (
    <>
      <StyledModal className={props.isActivated && "active"} onClick={() => props.SetActivated(false)}>
        <StyledContent onClick={(event) => event.stopPropagation()}>
          <StyledButton onClick={() => props.SetActivated(false)}>
            <IoCloseCircleOutline />
          </StyledButton>
        </StyledContent>
      </StyledModal>
    </>
  );
};

export { Modal };
