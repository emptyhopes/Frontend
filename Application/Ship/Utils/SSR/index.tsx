import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
