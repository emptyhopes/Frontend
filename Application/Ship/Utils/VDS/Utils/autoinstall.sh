#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
