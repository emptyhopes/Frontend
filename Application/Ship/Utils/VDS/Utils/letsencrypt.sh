#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname