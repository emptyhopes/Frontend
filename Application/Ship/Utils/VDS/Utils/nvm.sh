#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
