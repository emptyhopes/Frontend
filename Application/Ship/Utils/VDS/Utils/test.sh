#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru