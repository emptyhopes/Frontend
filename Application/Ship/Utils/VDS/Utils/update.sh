#! /bin/bash

sudo apt-get update --yes && sudo apt-get upgrade --yes
sudo apt-get autoremove --yes

sudo apt-get install git nginx certbot python3-certbot-nginx --yes