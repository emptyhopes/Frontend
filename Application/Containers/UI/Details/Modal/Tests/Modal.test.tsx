import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Modal } from "@/Application/Containers/UI/Details/Modal/Modal";

test("Modal", () => {
  const { container } = RenderTests(<Modal />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
