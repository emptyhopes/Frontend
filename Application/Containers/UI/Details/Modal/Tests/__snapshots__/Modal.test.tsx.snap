// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100vw;
  height: 100vh;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: rgba(0,0,0,0.8);
  opacity: 0;
  pointer-events: none;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0.active {
  opacity: 1;
  pointer-events: all;
}

.c1 {
  position: relative;
  width: 800px;
  height: 600px;
  border: 1px solid #000000;
  border-radius: 20px;
  background-color: #ffffff;
}

.c2 {
  position: absolute;
  top: 12px;
  right: 10px;
  width: 30px;
  height: 30px;
  border: none;
  background: transparent;
}

.c2 svg {
  width: 20px;
  height: 20px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        aria-label="modal-close"
        class="c2"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
            fill="none"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M320 320L192 192m0 128l128-128"
            fill="none"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
